engine/src/main/java/org/hibernate/validator/internal/util/ExecutableHelper.java

```diff
@@ -197,9 +211,20 @@ public static String getSignature(String name, Class<?>[] parameterTypes) {
 	 * @return A string representation of the given executable.
 	 */
 	public static String getExecutableAsString(String name, Class<?>... parameterTypes) {
-		return Stream.of( parameterTypes )
-			.map( t -> t.getSimpleName() )
-			.collect( Collectors.joining( ", ", name + "(", ")" ) );
+		StringBuilder signature = new StringBuilder( name.length() + 2 + parameterTypes.length * 25 );
+		signature.append( name ).append( '(' );
+		boolean separator = false;
+		for ( Class<?> parameterType : parameterTypes ) {
+			if ( separator ) {
+				signature.append( ", " );
```

Comment:
if the goal was merely uniqueness, you won't need the extra space and could still benefit from the single char invocation by just using ',' as separator.

Reply:
Yeah so that one is about having a nice string representation. That's why it's still there and is in pretty print mode.